### PR TITLE
Install emails pack

### PR DESCRIPTION
**Pull request by maximebiloe**
| Questions | Answers |
| --- | --- |
| Branch? | develop |
| Description? | From now, we have email packs for 1.7 and no more using of 1.6 packs. I've also updated englais emails. |
| Type? | improvement |
| Category? | CO |
| BC breaks? | no |
| Deprecations? | no |
| Fixed ticket? | http://forge.prestashop.com/browse/BOOM-1489 |
| How to test? | Install in different languages and check for the presence of the right folder into `mails`. It should be okay when you import a language from your backoffice too. And finally, you should be able to change the order state when you're using french for example. |

### CODE REVIEW

File: classes/Language.php [hunk 1]
```diff
@@ -28,8 +28,8 @@
 class LanguageCore extends ObjectModel
 {
     const ALL_LANGUAGES_FILE = '/app/Resources/all_languages.json';
-    const LANGUAGE_GZIP_URL = 'http://translations.prestashop.com/download/lang_packs/gzip/%s/%s.gzip';
     const SF_LANGUAGE_PACK_URL = 'http://translate.prestashop.com/TEMP/TEMP/TEMP/TEMP/TEMP/%s.zip';
+    const EMAILS_LANGUAGE_PACK_URL = 'http://translate.prestashop.com/TEMP/TEMP/TEMP/TEMP/emails/%s.zip';
```
Comment (PrestaEdit) [inline classes/Language.php]: "TEMP/TEMP/TEMP/TEMP/" :laughing:

Comment (maximebiloe) [inline classes/Language.php]: Yeah we know. We are preparing the server to host the translation packs but it's not ready yet

File: classes/Language.php [hunk 2]
```diff
@@ -990,43 +976,61 @@ public static function installSfLanguagePack($locale, &$errors = array())
...
-                    $mails_files[] = $data;
+            foreach ($coreMails as $coreEmail) {
+                if ('.' !== $coreEmail && '..' !== $coreEmail) {
+                    rename($folder.'/core/'.$coreEmail, $coreDestPath.'/'.$coreEmail);
                 }
             }
 
-            $files_list = array_diff($files_list, $mails_files);
-        }
+            $modules = scandir($folder.'/modules');
```
Comment (mickaelandrieu) [inline classes/Language.php]: really ? can you use Finder component or `DirectoryIterator` please ?

```php
$finder = \Symfony\Component\Finder\Finder::create();
```

File: classes/Language.php [hunk 3]
```diff
@@ -990,43 +976,61 @@ public static function installSfLanguagePack($locale, &$errors = array())
...
+                    $moduleMails = array_merge(
+                        Tools::scandir($folder.'/modules/'.$module, 'html', '', true),
+                        Tools::scandir($folder.'/modules/'.$module, 'txt', '', true)
+                    );
+
+                    if (!is_dir($moduleDestPath)) {
+                        mkdir($moduleDestPath, 0755, true);
+                    }
+                    foreach ($moduleMails as $moduleEmail) {
+                        rename(
```
Comment (mickaelandrieu) [inline classes/Language.php]: same for rename and mkdir stuff use Filesystem component :)

File: classes/Language.php [hunk 4]
```diff
@@ -990,43 +976,61 @@ public static function installSfLanguagePack($locale, &$errors = array())
...
+                            $folder.'/modules/'.$module.'/'.$moduleEmail,
+                            $moduleDestPath.'/'.$moduleEmail
+                        );
+                    }
+                }
+            }
 
-        if (!$gz->extractList(AdminTranslationsController::filesListToPaths($files_list), _PS_TRANSLATIONS_DIR_.'../')) {
-            $errors[] = sprintf(Tools::displayError('Cannot decompress the translation file for the following language: %s'), (string) $iso);
+            Tools::deleteDirectory($folder);
```
Comment (mickaelandrieu) [inline classes/Language.php]: same (Filesyystem)

**Review by mickaelandrieu — CHANGES_REQUESTED** [review]
More Symfony, less Legacy my friend :)

**Comment by maximebiloe** [comment]
@mickaelandrieu Thanks for the review. I've made the modifications

File: classes/Language.php [hunk 5]
```diff
@@ -988,38 +990,31 @@ public static function installEmailsLanguagePack($lang_pack, &$errors = array())
...
-                            $folder.'/modules/'.$module.'/'.$moduleEmail,
-                            $moduleDestPath.'/'.$moduleEmail
+            if ($fileSystem->exists($folder.'/modules')) {
+                foreach ($finder->directories()->in($folder.'/modules') as $moduleDirectory) {
+                    $moduleDestPath = _PS_ROOT_DIR_.'/modules/'.$moduleDirectory->getFileName().'/mails/'.$lang_pack['iso_code'];
+                    $fileSystem->mkdir($moduleDestPath, 0755);
+
+                    $findEmails = new \Symfony\Component\Finder\Finder();
+                    foreach ($findEmails->files()->in($moduleDirectory->getRealPath()) as $moduleEmail) {
+                        dump($moduleEmail->getRealpath(), $moduleDestPath.'/'.$moduleEmail->getFileName());
```
Comment (mickaelandrieu) [inline classes/Language.php]: dump ? :)

Comment (maximebiloe) [inline classes/Language.php]: 😊

Comment (maximebiloe) [inline classes/Language.php]: I don't know what you're talking about ^^

**Comment by aleeks** [comment]
Goood gooood gooooood job !!!!! Thanks @maximebiloe

**Pull request merged**